exonum/src/storage/base_index.rs

```diff
@@ -52,25 +55,59 @@ pub struct BaseIndexIter<'a, K, V> {
 }
 
 impl<T> BaseIndex<T> {
-    /// Creates a new index representation based on the common prefix of its keys and storage view.
+    /// Creates a new index representation based on the name and storage view.
     ///
     /// Storage view can be specified as [`&Snapshot`] or [`&mut Fork`]. In the first case only
     /// immutable methods are available. In the second case both immutable and mutable methods are
     /// available.
     /// [`&Snapshot`]: ../trait.Snapshot.html
     /// [`&mut Fork`]: ../struct.Fork.html
-    pub fn new(prefix: Vec<u8>, view: T) -> Self {
-        BaseIndex {
-            prefix: prefix,
-            view: view,
+    pub fn new(name: &str, view: T) -> Self {
+        if is_valid_name(name) {
+            BaseIndex {
+                name: name.to_string(),
+                prefix: None,
+                view,
+            }
+        } else {
+            panic!("Wrong characters using in name. Use: a-zA-Z0-9 and _");
+        }
+    }
+
+    /// Creates a new index representation based on the name, common prefix of its keys
+    /// and storage view.
+    ///
+    /// Storage view can be specified as [`&Snapshot`] or [`&mut Fork`]. In the first case only
+    /// immutable methods are available. In the second case both immutable and mutable methods are
+    /// available.
+    /// [`&Snapshot`]: ../trait.Snapshot.html
+    /// [`&mut Fork`]: ../struct.Fork.html
+    pub fn with_prefix(name: &str, prefix: Vec<u8>, view: T) -> Self {
```

Comment:
Why not `AsRef<str>`?

Reply:
For a name I agree but for a prefix i guess it's needless. The prefix will change ownership and won't be used in another place.